### PR TITLE
Return 400 if file system is readonly on function PUTs

### LINKED ISSUE #7625: PUT /admin/functions/{funcName} should return 400 if file system is readonly
Check file system status using
https://github.com/Azure/azure-functions-host/blob/a90271a7430af9bb8ba0554461927151301226ad/src/WebJobs.Script/Config/ScriptApplicationHostOptions.cs#L35

Changes needed would be similar to how appoffline is implemented.

### PR DESCRIPTION
**Pull request by pgombar**
### Issue describing the changes in this PR

resolves #7625

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs [hunk 1]
```diff
@@ -67,6 +69,11 @@ public async Task<IActionResult> Get(string name)
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
         public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata, [FromServices] IFileMonitoringService fileMonitoringService)
         {
+            if (_applicationHostOptions.Value.IsFileSystemReadOnly)
+            {
+                return BadRequest("Filesystem is read-only");
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs]: nit: more descriptive error message - `Your app is currently in read only mode. Cannot create or update functions`.
For reference - Portal displays - `Your app is currently in read only mode because you are running from a package file. To make any changes update the content in your zip file and WEBSITE_RUN_FROM_PACKAGE app setting.` when function app content cannot be edited.

File: test/WebJobs.Script.Tests/Controllers/Admin/FunctionsControllerTests.cs [hunk 2]
```diff
@@ -73,9 +75,13 @@ public async Task Invoke_CallsFunction()
                 Input = testInput
             };
 
+            var scriptPath = Path.GetTempPath();
+            var applicationHostOptions = new ScriptApplicationHostOptions();
+            applicationHostOptions.ScriptPath = scriptPath;
+            var optionsWrapper = new OptionsWrapper<ScriptApplicationHostOptions>(applicationHostOptions);
             var functionsManagerMock = new Mock<IWebFunctionsManager>();
             var mockRouter = new Mock<IWebJobsRouter>();
-            var testController = new FunctionsController(functionsManagerMock.Object, mockRouter.Object, new LoggerFactory());
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests/Controllers/Admin/FunctionsControllerTests.cs:78]: I do not see a test for CreateOrUpdate. Please add one

**Comment by pragnagopa** [comment]
Please update the issue - with more details - specifically is this a recent regression or an improvement we would like to address as we now have single source of truth  to check file system is read-only?

**Review by pragnagopa — COMMENTED** [review]
Added couple of comments. Thanks!

**Comment by balag0** [comment]
> Please update the issue - with more details - specifically is this a recent regression or an improvement we would like to address as we now have single source of truth to check file system is read-only?

This is a repair item from an issue on the discussion alias - "Functions disappeared suddenly when they executing"
Not a regression but more customers can hit this as we support editing for more scenarios.

**Comment by pragnagopa** [comment]
Marked #7625 as a bug - thanks @balag0 !

**Comment by vrdmr** [comment]
Can you also update the Wiki - https://github.com/Azure/azure-functions-host/wiki/Admin-API to showcase this API and what it does? Thanks.

File: test/WebJobs.Script.Tests/Controllers/Admin/FunctionsControllerTests.cs [hunk 3]
```diff
@@ -85,6 +94,46 @@ public async Task Invoke_CallsFunction()
...
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Host = new HostString("local");
+            httpContext.Request.Method = "Put";
+            httpContext.Request.Path = $"/admin/functions/{testFunctionName}";
+            testController.ControllerContext.HttpContext = httpContext;
+
+            var fileMonitoringServiceMock = new Mock<IFileMonitoringService>();
+            var result = (ObjectResult)(await testController.CreateOrUpdate(testFunctionName, functionsMetadataMock.Object, fileMonitoringServiceMock.Object));
+
+            Assert.Equal((int)expected, result.StatusCode);
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests/Controllers/Admin/FunctionsControllerTests.cs]: assert error message as well. Also, extend the test to include other cases that can cause 400 - empty function name?

Comment (pgombar) [reply test/WebJobs.Script.Tests/Controllers/Admin/FunctionsControllerTests.cs]: thanks, updated!

**Review by pragnagopa — APPROVED** [review]
